http-core/src/main/scala/org/apache/pekko/http/scaladsl/model/Uri.scala

```diff
@@ -129,7 +129,8 @@ sealed abstract case class Uri(scheme: String, authority: Authority, path: Path,
   /**
    * Returns a copy of this Uri with the given query.
    */
-  def withQuery(query: Query): Uri = copy(rawQueryString = if (query.isEmpty) None else Some(query.toString))
+  def withQuery(query: Query): Uri =
+    createUnsafe(scheme, authority, path, if (query.isEmpty) None else Some(query.toString), fragment)
```

Comment:
Consider adding a `copyUnsafe` method and calling it here and also in all the other `with...` methods (this assumes that the query stored inside `Uri` is also already encoded).